src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5264,14 +5264,21 @@ private BoundExpression MakeMemberAccessValue(BoundExpression expr, DiagnosticBa
 
                 if (searchExtensionMethodsIfNecessary)
                 {
-                    return new BoundMethodGroup(
+                    var boundMethodGroup = new BoundMethodGroup(
                         node,
                         typeArguments,
                         boundLeft,
                         rightName,
                         lookupResult.Symbols.All(s => s.Kind == SymbolKind.Method) ? lookupResult.Symbols.SelectAsArray(s_toMethodSymbolFunc) : ImmutableArray<MethodSymbol>.Empty,
                         lookupResult,
                         flags);
+
+                    if (!boundMethodGroup.HasErrors && boundMethodGroup.ResultKind == LookupResultKind.Empty && typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument))
+                    {
+                        Error(diagnostics, ErrorCode.ERR_BadArity, node, rightName, MessageID.IDS_MethodGroup.Localize(), typeArgumentsSyntax.Count);
```

Comment:
From talking to Omar, the `Localize()` call is also used in other places where ERR_BadArity is reported.
Also, the `BindMemberAccessReportError` requires a `lookupError` parameter (a `DiagnosticInfo`), so it won't help.

#Closed